### ISSUE TITLE
Make water source forming filterable

### ISSUE THREAD

**Issue by RuedigerLP** [PGMDev/PGM#556]
so it can be controlled with applicators.

**Comment by Pablete1234** [comment]
Water sources are already filterable. It's a bug in bukkit that there's no event that reports the block transformation.

Example xml that should work:
```
<filters>
    <deny id="no-water-sources">
      <all>
        <material>8</material>
        <cause>WORLD</cause>
      </all>
    </deny>
</filters>
<regions>
    <apply block="no-water-sources">
        <region>
            <everywhere/>
        </region>
    </apply>
</regions>
```

Denies material 8 (water source) if it's caused by the world (means players can still place water with buckets or by breaking ice, but ice naturally melting is denied)

**Comment by AustinLMayes** [comment]
Bukkit event added [downstream](https://github.com/StratusNetwork/SportPaper/commit/04c34e36607789e9d8c63497d84d236a00e8a592). The SportPaper team is waiting for it to be PR'd up and I'll merge it in. Until then this functionality is not possible in PGM. This functionality will also only be available on servers running SportPaper. 

Leaving this open until the needed SportPaper changes are brought upstream.

**Comment by Electroid** [comment]
The issue is for SportPaper independence, I don't really want to add another SportPaper dependency. Can this be done without a new event (or is the event introduced in later Bukkit versions?)

**Comment by Pablete1234** [comment]
@KingOfSquares investigated on it, and apparently, no event was being thrown by bukkit for it. I think BlockPhysicsEvent should be receiving this, but i didn't test

**Comment by KingOfSquares** [comment]
> @KingOfSquares investigated on it, and apparently, no event was being thrown by bukkit for it. I think BlockPhysicsEvent should be receiving this, but i didn't test

Yep, shiny added an event for it on his fork somewhere, but not possible with just cancelling the `BlockPhysicsEvent`s
